Update Buildkite docker image and publish lint artifacts

oguzkocer opened the pull request:
This PR addresses a few immediate issues with the current Buildkite configuration in preparation of publishing to S3:
* Updates docker image tag to `v1.1.0`
* Removes the copy `gradle.properties-example` step which seems to have been removed
* Publishes lint & checkstyle artifacts

@wzieba @0nko It looks like there are quite a few lint/test failures in the current configuration. I am guessing this happened because the CI wasn't working properly, which actually happened due to a security change we made in AWS and didn't realize it affected this repo. Could you take a look at them before I tackle publishing to S3? Please let me know if you need any help fixing them.

oguzkocer committed:
Update Buildkite docker image and publish lint artifacts

0nko commented:
Awesome, thanks @oguzkocer!

> It looks like there are quite a few lint/test failures in the current configuration

The project has undergone some major changes and I focused mainly on the functionality implementation. I'm almost done and I'm going to look at the lint issues next.

0nko commented:
@oguzkocer All lint and ktlint issues should be resolved now. If you pull the changes from the `trunk` you should be able to build it without any errors.

oguzkocer committed:
Merge remote-tracking branch 'origin/trunk' into update/buildkite-configuration

oguzkocer commented:
@0nko Thank you for working on those issues! 🙇‍♂️ I've merged `trunk` and it looks like most of the issues are resolved, but there is still [one error and one warning (if you want to handle that as well) in the lint task](https://s3.amazonaws.com/buildkiteartifacts.com/b26f1746-2f0c-49cc-bbd1-a45a5e24c351/a996dd63-4bf0-4d51-a63f-638847834b80/aca3eb14-52ca-4f59-b60b-2e5b8c4a6d91/b4d4c10f-a965-48c8-8ecd-df7a2910a2ba/mediapicker/build/reports/lint-results-release.html?response-content-type=text%2Fhtml&X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=ASIAQPCP3C7L3Y3J6HLL%2F20211109%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20211109T145305Z&X-Amz-Expires=600&X-Amz-SignedHeaders=host&X-Amz-Security-Token=IQoJb3JpZ2luX2VjEC4aCXVzLWVhc3QtMSJIMEYCIQDNJIWF07fdvNAc4%2F50p4z2CIEg2rWTCGZkzqgCs7B0zQIhAKmBAp9Q%2BHiTSQCZOLOTRH7k%2B%2BqnJAACQFXRUyR1WWp%2FKoMECNf%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEQABoMMDMyMzc5NzA1MzAzIgyPF8HZUH5DMWuRHX4q1wOgN30EY1kFwfA%2FeDtcVEhPGnTO2rjwvT32YiPUTs4M8L%2BKOlsl2G%2BLKmJujSnAqP6%2FH0QU1KC9G%2BSTTls1GfqH1DxDTpWxymfp3OtIfJmfkxnlAFfCILGuMXIqzPvAC9aom0SWPnFTfiq9WSjR8VhEVbNJPWKwP2B%2BL1gL8E5QTbqSJ68G5AKleZBkVt5JXSg52790gAwLe3liCfEoft%2BiMuWeNkURekzkzdyOkIzl1%2FtxPkDM7%2FiPCFWinmO9RA2PSqt9UstWm%2BOoZzKBlKOXcZrbZ1d9pC%2B7kPnMucWAeIoa7wuQsHBqSn41TwCQUc6pGlk%2FuMSMfNxBRhx7ZD8CziIqFa9VkDzT5cGBUUFBZc2a7b51PEWWc2SOWVy1T6hzU291PHI7qveoRuQMbqHAm5VfDPvelVchj7DQ5ZhAxsAHZ8e0kZNFpuQlqH1m3BC8PCcQxxEWVx0tdEmL%2BdOc4uMG9uj%2BlUTeStNtKb8cct3B%2FoJdht3L%2BTdl%2FjLRyfdlR6bygfpGFHSxZFSk81gLllPa8g8NkRGnyvsjFmP5pt8xN9Dk9nltEVrJ9DcV5f76%2B2ma4d69e1Os%2FyI3M0BETMYG2wC0dObsc4B03bHjVrJuAV%2FEE3wwqvKpjAY6pAGz1IXi7QrX6oysWWW6%2BTm3fjtgQaj6r7aIaupHHUSfmjWjWizFrLTUwSXTrMBXD2n3orP0nc73AvdQkXUhZvGGNBUDFCQJLFQHGXTbCqSRCYl0JIiWt3bACwV66eok1EgSt0NDLqAmAouTI1c1pKL3Lvfk0PJ4WvUeSY%2FrqjGV5EBO%2FQzSKhAmDDktDhje2ZzBZspJdjgKyJQYQN5ncUnBKneG1g%3D%3D&X-Amz-Signature=51e485afc30552add534b725a7b12ffe8788b6f5573083759b9b7119c969ba2d).

_Edit: It looks like the link gets expired very quickly, so the artifact should be retrieved directly from the Buildkite job._

0nko commented:
@oguzkocer Strange, I'm not seeing any issues when I run the lint check locally. I saw this issue before but I've added a `lint.xml` to change the rules. It seems like buildkite is ignoring it. Do you know why?

oguzkocer committed:
Ignores IconMissingDensityFolder & GradleDependency lint issues

oguzkocer commented:
@0nko The error in the Buildkite lint task was related to `IconMissingDensityFolder` which wasn't in the `lint.xml` file. I've added that in f82895ace63a7baae74e7f1cd5b72cfca9ca0471 and that seems to have resolved it.

I had that error in my local run, and interestingly I also had the `GradleDependency` error which I also added to `lint.xml` as an `ignore` rule. It was complaining about us not using the latest version of utils library. With our s3 setup, unfortunately this can throw false-negative results, so it's best to ignore them.

The thing about issues popping up in different computers & Buildkite is always a bit confusing to me. I _think_ it might be related to the different `jdk` versions. We had the same thing in WordPress-Android, but after we addressed everything, we didn't have the issue again. It's something worth looking into in more detail in the future, but I think this setup is good enough right now if you're OK with it.

0nko commented:
Awesome, thank you @oguzkocer!

wzieba reviewed (approved):
This looks good to me, thanks for working on that @oguzkocer !
